### PR TITLE
Enable Collation tests

### PR DESCRIPTION
**Pull request by tarekgh**
this enable the tests for CompareInfo, SortKey and SortVersion

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @stephentoub @danmosemsft @weshaggard

File: src/System.Globalization/tests/CompareInfo/CompareInfoTests.netstandard1.7.cs [hunk 1]
```diff
@@ -297,17 +294,16 @@ public void SortKeyMiscTest()
             Assert.NotEqual(sk4.GetHashCode(), sk3.GetHashCode());
             Assert.NotEqual(sk4.KeyData, sk3.KeyData);
 
-            Assert.Equal(sk4, sk1);
-            Assert.Equal(sk4.GetHashCode(), sk1.GetHashCode());
-            Assert.Equal(sk4.KeyData, sk1.KeyData);
+            Assert.Equal(sk4, sk5);
+            Assert.Equal(sk4.GetHashCode(), sk5.GetHashCode());
+            Assert.Equal(sk4.KeyData, sk5.KeyData);
```
Comment (stephentoub) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.netstandard1.7.cs]: Do these fail on Unix without this change? If so, do we know why? And if it's by design, should we have a Windows-only test to verify the original case?

Comment (tarekgh) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.netstandard1.7.cs]: Actually the test assumption was wrong but it happened to work on Windows. the test was written with the assumption of equality of sort key generated from lowercase string with case sensitive option and a sort key generated from same string but with upper casing and case insensitive option. this is wrong assumption but it happened to have Windows returned same key in such case. we cannot depend on this assumption on Windows as we can run into cases to break this assumption there. 

the correct thing is to compare sort keys generated from lower and upper casing with the case insensitive option.

**Comment by tarekgh** [comment]
#11608 
#11614 
https://github.com/dotnet/coreclr/pull/7502

**Pull request merged**